pyannote/audio/interactive/recipes/voice_activity_detection.py

```diff
@@ -0,0 +1,222 @@
+    extend: float = 2.0,
+) -> Iterable[Dict]:
+    """
+    Stream for pyannote.voice_activity_detection recipe
+    Applies (pretrained) speech activity detection and sends the results for
+    manual correction chunk by chunk.
+    Parameters
+    ----------
+    pipeline : VoiceActivityDetection
+        Pretrained speech activity detection pipeline.
+    source : Path
+        Directory containing audio files to process.
+    chunk : float, optional
+        Duration of chunks, in seconds. Defaults to 10s.
+    Yields
+    ------
+    task : dict
+        Prodigy task with the following keys:
+        "path" : path to audio file
+        "text" : name of audio file
+        "chunk" : chunk start and end times
+        "audio" : base64 encoding of audio chunk
+        "audio_spans" : speech spans detected by pretrained SAD model
+        "audio_spans_original" : copy of "audio_spans"
+        "meta" : additional meta-data displayed in Prodigy UI
+    """
+    raw_audio = Audio(sample_rate=SAMPLE_RATE, mono=True)
+
+    for audio_source in AudioLoader(source):
+
+        path = audio_source["path"]
+        text = audio_source["text"]
+        file = {"uri": text, "database": source, "audio": path}
+
+        duration = raw_audio.get_duration(file)
+        file["duration"] = duration
+
+        if duration <= chunk:
+            if pipeline is not None:
+                speech: Annotation = pipeline(file)
+                task_audio_spans = to_audio_spans(speech)
+            else:
+                task_audio_spans = []
+            waveform, sr = raw_audio.crop(file, Segment(0, duration))
+            waveform = waveform.numpy().T
+            task_audio = to_base64(normalize(waveform), sample_rate=SAMPLE_RATE)
+
+            yield {
+                "path": path,
+                "text": text,
+                "audio": task_audio,
+                "audio_spans": task_audio_spans,
+                "audio_spans_original": deepcopy(task_audio_spans),
+                "chunk": {"start": 0, "end": duration},
+                "meta": {"file": text},
+            }
+
+        else:
+            for focus in chunks(duration, chunk=chunk, shuffle=False):
+                task_text = f"{text} [{focus.start:.1f}, {focus.end:.1f}]"
+                waveform, sr = raw_audio.crop(file, focus)
+                if waveform.shape[1] != SAMPLE_RATE * chunk:
+                    waveform = waveform.pad(
+                        input=waveform,
+                        pad=(0, SAMPLE_RATE * chunk - waveform.shape[1]),
+                        mode="constant",
+                        value=0,
+                    )
+                if pipeline is not None:
+                    st = focus.start - extend
+                    ed = focus.end + extend
+                    if focus.start == 0:
+                        st = 0
+                    elif focus.duration < chunk:
+                        ed = focus.end
+                    longFocus = Segment(st, ed)
```

Comment:
Est-ce que ceci ne serait pas plus lisible ?

```suggestion
                    longFocus = Segment(max(0, focus.start - extend), 
                                        min(focus.end + extend, duration))
```